### PR TITLE
Python Generate OpCreation Methods by OpProto

### PR DESCRIPTION
**Pull request by reyoung**
All OpCreation method are generated by
`create_op_creation_methods::__bootstrap__` method, and stores in
`op_creations` object and its methods.

There are three parts to implement this feature.

1. Get all registered `OpProto` from C++ side. It is implemented in
`get_all_op_protos` method.
1. Create a function to convert `kwargs` to `OpDesc` base on each op's
`OpProto`. The `OpDescCreationMethod` class.
1. Convert `OpProto` to `docstring` by `get_docstring_from_op_proto`
method.

All three methods are unit tested. The `__bootstrap__` just combines
them together and create a method in runtime.

For details, please reference the doc string in
`create_op_creation_methods.py` and the unit test
`test_op_creation_methods.py`.

### CODE REVIEW

File: paddle/framework/op_registry.h [hunk 1]
```diff
@@ -222,6 +235,17 @@ class OpRegistry {
   };
 
  private:
+  static void AssignTempVariable(OperatorBase* op) {
```
Comment (jacquesqiao) [inline paddle/framework/op_registry.h]: Assign is not a good name, maybe GenNameForTempVar?

Comment (reyoung) [inline paddle/framework/op_registry.h]: Cool!

File: paddle/framework/op_registry.h [hunk 2]
```diff
@@ -199,19 +200,30 @@ class OpRegistry {
...
               std::back_inserter(op->outputs_));
+
+    //! Fill attrs, and validate attrs.
     for (auto& attr : op_desc.attrs()) {
       op->attrs_[attr.name()] = AttrTypeHelper::GetAttrValue(attr);
     }
     op_checkers().at(op_type).Check(op->attrs_);
+
+    //! Convert Temporary variable name to an unique variable name.
+    GenerateTempVariableName(op.get());
```
Comment (jacquesqiao) [inline paddle/framework/op_registry.h]: 因为temp var是c++这边自动生成的，所以python那边就管不了这些Var了是么，这些Var应该如被谁创建？

File: paddle/framework/operator.h [hunk 3]
```diff
@@ -39,6 +39,13 @@ using OperatorPtr = std::shared_ptr<OperatorBase>;
  */
 class OperatorBase {
  public:
+  /// If a variable is a empty variable, that name will be used.
```
Comment (jacquesqiao) [inline paddle/framework/operator.h]: should add some more comment about what is `empty variable` and `temp variable`

File: paddle/framework/op_registry.h (hunk 2 above)
Comment (reyoung) [inline paddle/framework/op_registry.h]: Python这边会在运行Op之前扫描这个Op的输入和输出，进而创建对应的variable

File: python/paddle/v2/framework/create_op_creation_methods.py [hunk 4]
```diff
@@ -1,11 +1,246 @@
...
+    """
     protostrs = core.get_all_op_protos()
     ret_values = []
     for pbstr in protostrs:
         op_proto = op_proto_pb2.OpProto.FromString(str(pbstr))
         ret_values.append(op_proto)
     return ret_values
+
+
+class OpDescCreationMethod(object):
```
Comment (jacquesqiao) [inline python/paddle/v2/framework/create_op_creation_methods.py]:
```python
class OpDescCreator(object):
  des create(self, *args, **kwargs):
    ...
    return op_desc     
``

I think above is better for reading and understanding code.

Comment (reyoung) [inline python/paddle/v2/framework/create_op_creation_methods.py]: A `Function object`(https://en.wikipedia.org/wiki/Function_object) or functor is a very common way for many programming languages.

By using `Function object`, can hold internal states and make function implementation into small pieces and more readable.

File: python/paddle/v2/optimizer.py [hunk 5]
```diff
@@ -17,6 +16,7 @@
 
 class Optimizer(object):
     def __init__(self, **kwargs):
+        import py_paddle.swig_paddle as swig_api
```
Comment (jacquesqiao) [inline python/paddle/v2/optimizer.py]: what is the purpose of this change? to speed up the importing?

Comment (reyoung) [inline python/paddle/v2/optimizer.py]: The same reason in #2864

File: python/paddle/v2/framework/tests/test_op_creation_methods.py [hunk 6]
```diff
@@ -1,15 +1,254 @@
...
+:type temp output: basestr
+:param str_attr: A string attribute for test op
+:type str_attr: basestr
+'''
+        self.assertEqual(expected_docstring, actual)
+
+
+class TestOpCreations(unittest.TestCase):
+    def test_all(self):
+        add_op = creation.op_creations.add_two(X="a", Y="b", Out="z")
```
Comment (jacquesqiao) [inline python/paddle/v2/framework/tests/test_op_creation_methods.py]: creation.op_creations 感觉应该做一个alias
比如 ops  或者  operators，不过这个可以后边再搞
